go/mcap/indexed_message_iterator.go

```diff
@@ -96,11 +94,17 @@ func (it *indexedMessageIterator) parseSummarySection() error {
 			if err != nil {
 				return fmt.Errorf("failed to parse attachment index: %w", err)
 			}
+			it.chunkIndexes = append(it.chunkIndexes, idx)
```

Comment:
why move this up here?

Reply:
or if I understand right, is it needed at all? I guess this allows reuse of the reader in a way, since it would then have the full set of chunk indexes - is that the point?

Reply:
I can revert this change. Previously i thought since `Info()` returns the chunkIndexes from here, it was a bug that the chunkIndexes in Info depended on the time / topic filter you passed in. However `Info()` constructs its own message iterator with no filters, so it's not a bug.